model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TypeInfoTests.xtend

```diff
@@ -14,1049 +15,1552 @@ import hu.blackbelt.judo.meta.jsl.jsldsl.EntityDerivedDeclaration
-
-		assertEquals(TypeInfo.PrimitiveType.BOOLEAN, TypeInfo.getTargetType(typeOfField.expression).getPrimitive)
-		assertEquals(TypeInfo.PrimitiveType.BOOLEAN, TypeInfo.getTargetType(kindOfField.expression).getPrimitive)
-		
-		val TypeInfo containerTypeInfo = TypeInfo.getTargetType(containerField.expression)
-		assertEquals(testEntity, containerTypeInfo.getEntity)
-		assertEquals(false, containerTypeInfo.isCollection)		
-
-		val TypeInfo asTypeTypeInfo = TypeInfo.getTargetType(asTypeField.expression)
-		assertEquals(t1Field.referenceType, asTypeTypeInfo.getEntity)
-		assertEquals(false, containerTypeInfo.isCollection)		
-
-		val memberOfField = testEntity.memberByName("memberOf") as EntityDerivedDeclaration
-		assertEquals(TypeInfo.PrimitiveType.BOOLEAN, TypeInfo.getTargetType(memberOfField.expression).getPrimitive)
-
-	}
-
-	@Test
-	def void testSelectorFunctions() {
-		val m = '''
-			model TestModel;
-			
-			type string String(min-size = 0, max-size = 32);
-
-			entity T1 {
-				field String name;
-			}
-
-			entity Test {
-				field T1[] t1s;
-				field T1 t1;
-
-				derived T1[] head => self.t1s!first(t | t.name);
-				derived T1[] tail => self.t1s!last(t | t.name);
-				derived T1 any => self.t1s!any();
-				derived Integer size => self.t1s!size();
-				derived T1[] asCollection => self.t1!asCollection(entityType = T1);
-				derived Boolean contains => self.t1s!contains(instance = self.t1);
-
-			}
-		'''.parse.fromModel
-	
-		val testEntity = m.entityByName("Test") 
-		val t1sField = testEntity.memberByName("t1s") as EntityFieldDeclaration		
-
-		val headField = testEntity.memberByName("head") as EntityDerivedDeclaration		
-		val tailField = testEntity.memberByName("tail") as EntityDerivedDeclaration		
-		val anyField = testEntity.memberByName("any") as EntityDerivedDeclaration		
-		val sizeField = testEntity.memberByName("size") as EntityDerivedDeclaration		
-		val asCollectionField = testEntity.memberByName("asCollection") as EntityDerivedDeclaration		
-		val containsField = testEntity.memberByName("contains") as EntityDerivedDeclaration		
-		
-		val TypeInfo headTypeInfo = TypeInfo.getTargetType(headField.expression)
-		assertEquals(t1sField.referenceType, headTypeInfo.getEntity)
-		assertEquals(true, headTypeInfo.isCollection)		
-		
-		val TypeInfo tailTypeInfo = TypeInfo.getTargetType(tailField.expression)
-		assertEquals(t1sField.referenceType, tailTypeInfo.getEntity)
-		assertEquals(true, tailTypeInfo.isCollection)		
-
-		val TypeInfo anyTypeInfo = TypeInfo.getTargetType(anyField.expression)
-		assertEquals(anyField.referenceType, anyTypeInfo.getEntity)
-		assertEquals(false, anyTypeInfo.isCollection)		
-
-		assertEquals(TypeInfo.PrimitiveType.NUMERIC, TypeInfo.getTargetType(sizeField.expression).getPrimitive)
-
-		val TypeInfo asCollectionTypeInfo = TypeInfo.getTargetType(anyField.expression)
-		assertEquals(asCollectionField.referenceType, asCollectionTypeInfo.getEntity)
-		assertEquals(false, anyTypeInfo.isCollection)
```

Comment:
If `asCollection` is re-added, this should be fixed.

Reply:
val TypeInfo asCollectionTypeInfo = TypeInfo.getTargetType(asCollectionField.expression)
        assertEquals(asCollectionField.referenceType, asCollectionTypeInfo.getEntity)
        assertEquals(true, asCollectionTypeInfo.isCollection)